graphbolt/src/cuda/common.h

```diff
@@ -88,6 +91,54 @@ inline bool is_zero<dim3>(dim3 size) {
     }                                                                 \
   }
 
+/**
+ * @brief This class is designed to handle the copy operation of a single
+ * scalar_t item from a given CUDA device pointer. Later, if the object is cast
+ * into scalar_t, the value can be read.
+ *
+ * auto num_edges = cuda::ReadScalar(indptr.data_ptr<scalar_t>() +
+ * indptr.size(0) - 1);
+ * // Perform many operations here, they will run as normal.
+ * // We finally need to read num_edges.
+ * auto indices = torch::empty(static_cast<scalar_t>(num_edges));
+ */
+template <typename scalar_t>
+struct ReadScalar {
```

Comment:
ReadScalar is a pretty bad name here, how come cudaMemcpyAsync happens in a method named read? Can you come up with a better name?

Reply:
Changed to CopyScalar, in the code, it is used as: `cuda::CopyScalar`.